### PR TITLE
PS-8556 Jenkins MTR run should also allow to run tests requiring ps-p…

### PR DESCRIPTION
**Pull request by surbhat1595**
…rotocol

### CODE REVIEW

**Review by EvgeniyPatlan — APPROVED** [review]

File: jenkins/pipeline-parallel-mtr.groovy [hunk 1]
```diff
@@ -1,4 +1,4 @@
-PIPELINE_TIMEOUT = 24
+, env.WITH_PS_PROTOCOL?.trim() == 'yes'PIPELINE_TIMEOUT = 24
```
Comment (inikep) [inline jenkins/pipeline-parallel-mtr.groovy]: There is something wrong here.

**Comment by inikep** [comment]
At https://github.com/Percona-Lab/ps-build/pull/403/files you also modified the following files:
```
docker/run-test
docker/run-test-parallel-mtr
jenkins/pipeline-parallel-mtr.yml
local/test-binary
local/test-binary-parallel-mtr
```
Why don't you just `git cherry-pick b94b3f70b5ceee2546` from `8.0` on top of `Percona-Lab:8.x` branch?

**Review by inikep — CHANGES_REQUESTED** [review]

**Comment by surbhat1595** [comment]
> At https://github.com/Percona-Lab/ps-build/pull/403/files you also modified the following files:
> 
> ```
> docker/run-test
> docker/run-test-parallel-mtr
> jenkins/pipeline-parallel-mtr.yml
> local/test-binary
> local/test-binary-parallel-mtr
> ```
> 
> Why don't you just `git cherry-pick b94b3f70b5ceee2546` from `8.0` on top of `Percona-Lab:8.x` branch?

@inikep Sorry, I forgot to add the changes made to the remaining files. Could you please review now?

**Comment by inikep** [comment]
`Percona-Lab:8.x` was recently forked from `Percona-Lab:8.0`.
That is why you should port your commit from `surbhat1595:8.0`, that is b94b3f70b5ceee2546.
Try `git cherry-pick b94b3f70b5ceee2546` on `surbhat1595:8.x` (this branch) and it will show you what file you are still missing.

**Comment by surbhat1595** [comment]
> `Percona-Lab:8.x` was recently forked from `Percona-Lab:8.0`. That is why you should port your commit from `surbhat1595:8.0`, that is [b94b3f7](https://github.com/Percona-Lab/ps-build/commit/b94b3f70b5ceee25465d0be3f0798342726295ca). Try `git cherry-pick b94b3f70b5ceee2546` on `surbhat1595:8.x` (this branch) and it will show you what file you are still missing.

@inikep Yes, I tried cherry-picking the commit from 8.0, but there were a lot of merge conflicts in various files due to the removal of ZEN_FS which would have taken a longer time to resolve. Hence, made the changes manually.

**Review by inikep — CHANGES_REQUESTED** [review]
OK, but the changes in `jenkins/pipeline-parallel-mtr.yml` are missing in this PR.

**Comment by surbhat1595** [comment]
> OK, but the changes in `jenkins/pipeline-parallel-mtr.yml` are missing in this PR.

@inikep I see that there are no other jenkins MTR configuration options present in https://github.com/Percona-Lab/ps-build/blob/8.x/jenkins/pipeline-parallel-mtr.yml. Do we still make changes to this file?

**Comment by inikep** [comment]
> > OK, but the changes in `jenkins/pipeline-parallel-mtr.yml` are missing in this PR.
> 
> @inikep I see that there are no other jenkins MTR configuration options present in https://github.com/Percona-Lab/ps-build/blob/8.x/jenkins/pipeline-parallel-mtr.yml. Do we still make changes to this file?

OK, I got it now. @oleksandr-kachan If everything works fine without these options in `jenkins/pipeline-parallel-mtr.yml` we should also remove them in `8.0`.

**Review by inikep — APPROVED** [review]
LGTM

**Review by inikep — CHANGES_REQUESTED** [review]
I just found some issues in `percona-server-8.0-pipeline-parallel-mtr`:
```
+ ln -s /tmp/ps_disk_dir/var /tmp/results/PS/mysql-test/var_ps_protocol
+ rsync -a -L --prune-empty-dirs --include '*/' --include '*.err' --exclude '*' /tmp/results/PS/mysql-test/var_ps_protocol /tmp/results/PS/mysql-test/mtr_var
rsync: [sender] link_stat "/tmp/results/PS/mysql-test/var_ps_protocol" failed: No such file or directory (2)
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1330) [sender=3.2.3]
```

File: local/test-binary-parallel-mtr [hunk 2]
```diff
@@ -330,6 +330,56 @@ if [[ "${CI_FS_MTR}" = 'yes' ]]; then
...
+        --parallel=$PARALLEL \
+        --ps-protocol \
+        --testcase-timeout=${TESTCASE_TIMEOUT} \
+        ${MTR_ARGS_PS} \
+        ${PS_TESTS} \
+        ${ANALYZER_MTR_ARGS} \
+        --junit-output=${WORKDIR_ABS}/junit_ps_protocol.xml \
+        --junit-package="${DOCKER_OS}.${CMAKE_BUILD_TYPE}.ps_protocol" || true
+
+        ln -s /tmp/ps_disk_dir/var $PWD/var_ps_protocol
```
Comment (inikep) [inline local/test-binary-parallel-mtr]: Please change `ln -s /tmp/ps_disk_dir/var $PWD/var_ps_protocol` to `ln -s $PWD/var $PWD/var_ps_protocol`.
Please also change formatting e.g. `ln -s` should be in the same line as `MTR_BUILD_THREAD=auto` as we did in https://github.com/Percona-Lab/ps-build/pull/409/files

**Review by inikep — APPROVED** [review]
LGTM

**Pull request merged**